### PR TITLE
V14 QA added user permission tests

### PR DESCRIPTION
**Pull request by andr317c**
This PR focuses on E2E testing the different permission for a user

### CODE REVIEW

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Users/Permissions/ContentStartNodes.spec.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  password: 'verySecurePassword123',
+};
+
+const userGroupName = 'TestUserGroup';
+
+const rootDocumentTypeName = 'RootDocumentType';
+const childDocumentTypeOneName = 'ChildDocumentTypeOne';
+const childDocumentTypeTwoName = 'ChildDocumentTypeTwo';
+let childDocumentTypeOneId = null;
+let childDocumentTypeTwoId = null;
```
Comment (nhudinh0309) [inline tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Users/Permissions/ContentStartNodes.spec.ts:16]: No need to define this variable here as it only uses in the beforeEach section.

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Users/Permissions/UICulture.spec.ts [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+const userGroupName = 'TestUserGroup';
+
+let testUserCookieAndToken = {cookie: "", accessToken: "", refreshToken: ""};
+let userGroupId = null;
+
+test.beforeEach(async ({umbracoApi}) => {
+  await umbracoApi.user.ensureNameNotExists(testUser.name);
+  await umbracoApi.userGroup.ensureNameNotExists(userGroupName);
+
```
Comment (nhudinh0309) [inline tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Users/Permissions/UICulture.spec.ts:17]:
```suggestion
```

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Users/Permissions/UICulture.spec.ts [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  userGroupId = await umbracoApi.userGroup.createSimpleUserGroupWithContentSection(userGroupName);
+});
+
+test.afterEach(async ({umbracoApi}) => {
+  // Ensure we are logged in to admin
+  await umbracoApi.loginToAdminUser(testUserCookieAndToken.cookie, testUserCookieAndToken.accessToken, testUserCookieAndToken.refreshToken);
+
```
Comment (nhudinh0309) [inline tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Users/Permissions/UICulture.spec.ts:24]:
```suggestion
```

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Users/Permissions/MediaStartNodes.spec.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+  childFolderOneId = await umbracoApi.media.createDefaultMediaFolderAndParentId(childFolderOneName, rootFolderId);
+  await umbracoApi.media.createDefaultMediaFolderAndParentId(childFolderTwoName, rootFolderId);
+
+  userGroupId = await umbracoApi.userGroup.createUserGroupWithMediaSection(userGroupName);
+});
+
+test.afterEach(async ({umbracoApi}) => {
+  // Ensure we are logged in to admin
+  await umbracoApi.loginToAdminUser(testUserCookieAndToken.cookie, testUserCookieAndToken.accessToken, testUserCookieAndToken.refreshToken);
+
```
Comment (nhudinh0309) [inline tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Users/Permissions/MediaStartNodes.spec.ts:37]:
```suggestion
```

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Users/Permissions/ContentStartNodes.spec.ts [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+let userGroupId = null;
+
+test.beforeEach(async ({umbracoApi}) => {
+  await umbracoApi.documentType.ensureNameNotExists(rootDocumentTypeName);
+  await umbracoApi.documentType.ensureNameNotExists(childDocumentTypeOneName);
+  await umbracoApi.documentType.ensureNameNotExists(childDocumentTypeTwoName);
+  await umbracoApi.user.ensureNameNotExists(testUser.name);
+  await umbracoApi.userGroup.ensureNameNotExists(userGroupName);
+
```
Comment (nhudinh0309) [inline tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Users/Permissions/ContentStartNodes.spec.ts:35]:
```suggestion
```

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Users/Permissions/ContentStartNodes.spec.ts [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+  childDocumentTypeOneId = await umbracoApi.documentType.createDefaultDocumentType(childDocumentTypeOneName);
+  childDocumentTypeTwoId = await umbracoApi.documentType.createDefaultDocumentType(childDocumentTypeTwoName);
+  rootDocumentTypeId = await umbracoApi.documentType.createDocumentTypeWithAllowedTwoChildNodes(rootDocumentTypeName, childDocumentTypeOneId, childDocumentTypeTwoId);
+  rootDocumentId = await umbracoApi.document.createDefaultDocument(rootDocumentName, rootDocumentTypeId);
+  childDocumentOneId = await umbracoApi.document.createDefaultDocumentWithParent(childDocumentOneName, childDocumentTypeOneId, rootDocumentId);
+  await umbracoApi.document.createDefaultDocumentWithParent(childDocumentTwoName, childDocumentTypeTwoId, rootDocumentId);
+
```
Comment (nhudinh0309) [inline tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Users/Permissions/ContentStartNodes.spec.ts:42]:
```suggestion
```

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Users/Permissions/MediaStartNodes.spec.ts [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+const childFolderOneName = 'ChildFolderOne';
+const childFolderTwoName = 'ChildFolderTwo';
+
+test.beforeEach(async ({umbracoApi}) => {
+  await umbracoApi.user.ensureNameNotExists(testUser.name);
+  await umbracoApi.userGroup.ensureNameNotExists(userGroupName);
+  await umbracoApi.media.ensureNameNotExists(rootFolderName);
+  await umbracoApi.media.ensureNameNotExists(childFolderOneName);
+  await umbracoApi.media.ensureNameNotExists(childFolderTwoName);
+
```
Comment (nhudinh0309) [inline tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Users/Permissions/MediaStartNodes.spec.ts:26]:
```suggestion
```

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Users/Permissions/MediaStartNodes.spec.ts [hunk 8]
```diff
@@ -0,0 +1,91 @@
...
+  rootFolderId = await umbracoApi.media.createDefaultMediaFolder(rootFolderName);
+  childFolderOneId = await umbracoApi.media.createDefaultMediaFolderAndParentId(childFolderOneName, rootFolderId);
+  await umbracoApi.media.createDefaultMediaFolderAndParentId(childFolderTwoName, rootFolderId);
+
```
Comment (nhudinh0309) [inline tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Users/Permissions/MediaStartNodes.spec.ts:30]:
```suggestion
```

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Users/Permissions/ContentStartNodes.spec.ts [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+  userGroupId = await umbracoApi.userGroup.createSimpleUserGroupWithContentSection(userGroupName);
+});
+
+test.afterEach(async ({umbracoApi}) => {
+  // Ensure we are logged in to admin
+  await umbracoApi.loginToAdminUser(testUserCookieAndToken.cookie, testUserCookieAndToken.accessToken, testUserCookieAndToken.refreshToken);
+  await umbracoApi.documentType.ensureNameNotExists(rootDocumentTypeName);
```
Comment (nhudinh0309) [inline tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Users/Permissions/ContentStartNodes.spec.ts:49]: Add steps to delete the input documents also.

**Review by nhudinh0309 — CHANGES_REQUESTED** [review]
Had some comments

File: tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Users/Permissions/ContentStartNodes.spec.ts (hunk 9 above)
Comment (andr317c) [reply tests/Umbraco.Tests.AcceptanceTest/tests/DefaultConfig/Users/Permissions/ContentStartNodes.spec.ts:49]: The Documents are deleted when I delete the DocumentTypes

**Pull request merged**